src/main/java/uk/gov/ch/developer/docs/controller/developer/SignInController.java

```diff
@@ -9,217 +9,143 @@
     }
 
     @GetMapping
-    public void getSignIn(HttpServletRequest httpServletRequest,
-            HttpServletResponse httpServletResponse) throws IOException {
+    public void getSignIn(final HttpServletRequest httpServletRequest,
+            final HttpServletResponse httpServletResponse) throws IOException {
 
-        Session chSession = (Session) httpServletRequest
+        final Session chSession = (Session) httpServletRequest
                 .getAttribute(SessionHandler.CHS_SESSION_REQUEST_ATT_KEY);
-
         // Redirect for user authentication (no scope specified)
-        redirectForAuth(chSession, httpServletRequest, httpServletResponse, null, false);
+        redirectForAuth(chSession, httpServletRequest, httpServletResponse);
     }
 
     /**
      * Redirects to a URI for the user to authenticate themselves
-     * 
+     *
      * @param session The user's session, retrieved from context
-     * @param request
-     * @param response
-     * @param companyNumber The company number a user is viewing
-     * @param force Boolean determining whether to force re-authentication
-     * @throws IOException
      */
-    protected void redirectForAuth(Session session, HttpServletRequest request,
-            HttpServletResponse response, String companyNumber, boolean force) throws IOException {
+    protected void redirectForAuth(final Session session, final HttpServletRequest request,
+            final HttpServletResponse response)
+            throws IOException {
 
-        // Find the original requested url
-        StringBuilder originalRequestUrl = new StringBuilder(request.getRequestURL().toString());
-        String queryString = request.getQueryString();
-        if (queryString != null) {
-            originalRequestUrl.append("?").append(queryString);
-        }
+        final String originalRequestUrl = getRequestURL(request);
+        final String nonce = generateSessionNonce(session);
 
-        // Set the scope
-        String scope = null;
-        if (companyNumber != null) {
-            scope = OAUTH_COMPANY_SCOPE_PREFIX + companyNumber;
+        // Build oauth uri and redirect
+        final String state = oauth.oauth2EncodeState(originalRequestUrl, nonce, "content");
+        final String authoriseUri = identityProvider.getAuthorisationUrl(state);
+        response.sendRedirect(authoriseUri);
+    }
+
+    /**
+     * Constructs the authorisation URI with additional force and hint
+     *
+     * @param originalRequestUri Original URI from which to redirect
+     * @param scope Scope of the request
+     * @param email Email address from the session, empty if not present
+     * @return Authorisation URI
+     */
+    protected String createAuthoriseURIWithForceAndHint(final String originalRequestUri,
+            final String scope, final String nonce, final String email) {
+        if (email == null) {
+            LOGGER.debug("No email supplied");
+        }
+        if (email == null) {
```

Comment:
Logic check doesn't match Logger message. And is duplicate.

Reply:
fixed